File: v1/code/go/src/jvmgo/ch05/instructions/stack/pop.go
package stack

import "jvmgo/ch05/instructions/base"
import "jvmgo/ch05/rtda"

/*
bottom -> top
[...][c][b][a]
            |
            V
[...][c][b]
*/
// Pop the top operand stack value
type POP struct{ base.NoOperandsInstruction }

func (self *POP) Execute(frame *rtda.Frame) {
	stack := frame.OperandStack()
	stack.PopSlot()
}

/*
bottom -> top
[...][c][b][a]
         |  |
         V  V
[...][c]
*/
// Pop the top one or two operand stack values
type POP2 struct{ base.NoOperandsInstruction }

func (self *POP2) Execute(frame *rtda.Frame) {
	stack := frame.OperandStack()
	stack.PopSlot()
	stack.PopSlot()
}

File: v1/code/java/jvmgo_java/ch05/src/main/java/com/github/jvmgo/instructions/stack/pop.java
package com.github.jvmgo.instructions.stack;

import com.github.jvmgo.instructions.base.NoOperandsInstruction;
import com.github.jvmgo.rtda.Frame;

public class pop extends NoOperandsInstruction {
    @Override
    public int getOpCode() {
        return 0x57;
    }

    @Override
    public void execute(Frame frame) throws Exception {
        frame.getOperandStack().popSlot();
    }
}
